sklearn/model_selection/_subselect.py

```diff
@@ -0,0 +1,573 @@
+    >>> search = GridSearchCV(pipe, param_grid=param_grid, scoring="accuracy")
+    >>> search.fit(X, y)
+    >>> ss = Refitter(search.cv_results_)
+    >>> ss.fit(by_standard_error(sigma=1))
+    >>> refitted_index = ss.transform("reduce_dim__n_components")
+    >>> refitted_index
+
+    """
+
+    def __init__(self, cv_results_: Dict, scoring: str = "score"):
+        self.cv_results_ = cv_results_
+        self.scoring = scoring
+
+    def _get_splits(self):
+        # Extract subgrid corresponding to the scoring metric of interest
+        fitted_key_strings = "\t".join(list(self.cv_results_.keys()))
+        if not all(s in fitted_key_strings for s in ["split", "params", "mean_test"]):
+            raise TypeError(
+                "cv_results_ must be a dict of fitted GridSearchCV or RandomSearchCV"
+                " objects."
+            )
+
+        _splits = [
+            i
+            for i in list(self.cv_results_.keys())
+            if f"test_{self.scoring}" in i and i.startswith("split")
+        ]
+        if len(_splits) == 0:
+            raise KeyError(f"Scoring metric {self.scoring} not found in cv grid.")
+        else:
+            return _splits
+
+    @property
+    def _n_folds(self):
+        # Extract number of folds from cv_results_. Note that we cannot get this from
+        # the `n_splits_` attribute of the `cv` object because it is not exposed to the
+        # refit callable.
+        return len(
+            list(
+                set(
+                    [
+                        i.split("_")[0]
+                        for i in list(self.cv_results_.keys())
+                        if i.startswith("split")
+                    ]
+                )
+            )
+        )
+
+    @property
+    def _score_grid(self):
+        # Extract subgrid corresponding to the scoring metric of interest
+        return np.vstack([self.cv_results_[cv] for cv in self._get_splits()]).T
+
+    @property
+    def _cv_means(self):
+        # Calculate means of subgrid corresponding to the scoring metric of interest
+        return np.array(np.nanmean(self._score_grid, axis=1))
+
+    @property
+    def _lowest_score_idx(self):
+        # Return index of the lowest performing model
+        return np.nanargmin(self._cv_means)
+
+    @property
+    def _best_score_idx(self):
+        # Return index of the highest performing model
+        return np.nanargmax(self._cv_means)
+
+    def _apply_thresh(self, param: str, l_cutoff: float, h_cutoff: float) -> int:
+        if not any(param in x for x in self.cv_results_["params"][0].keys()):
+            raise KeyError(f"Parameter {param} not found in cv grid.")
+        else:
+            hyperparam = [
+                i for i in self.cv_results_["params"][0].keys() if i.endswith(param)
+            ][0]
+
+        # Mask out all grid columns that are outside the performance window
+        self.cv_results_[f"param_{hyperparam}"].mask = np.where(
```

Comment:
mutating a value passed from gridsearchcv seems a bad idea, as it creates side effects to the CV fitting operation. Rather, a copy should be made...

Reply:
Good suggestion. I've updated the `__init__` of `Refitter` to assign a copy of the `cv_results_` object to a new attribute on the class -- `cv_results_constrained_`. This initial motivation for not doing this all along was that when I had tried before to just assign `self.cv_results = cv_results_`, I started running into some minor issues with `Refitter`'s `__repr__()`, and relatedly errors on GridSearchCV when `cv_results_` was no longer initialized with that same name exactly as an attribute. To your point, all the more reason to be on the safe side and just leave any mutation for the copy.